docstring/src/main/java/io/cucumber/docstring/DocStringTypeRegistry.java

```diff
@@ -5,31 +5,37 @@
 import java.lang.reflect.Type;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
 
 import static java.lang.String.format;
 
 @API(status = API.Status.STABLE)
 public final class DocStringTypeRegistry {
 
-    private final Map<String, DocStringType> docStringTypesByContentType = new HashMap<>();
-    private final Map<Type, DocStringType> docStringTypesByType = new HashMap<>();
+    private static final Class<String> DEFAULT_TYPE = String.class;
+    private static final String DEFAULT_CONTENT_TYPE = "";
+    private final Map<String, Map<Type, DocStringType>> docStringTypes = new HashMap<>();
 
     public DocStringTypeRegistry() {
-        defineDocStringType(new DocStringType(String.class, "", (String docString) -> docString));
+        defineDocStringType(new DocStringType(DEFAULT_TYPE, DEFAULT_CONTENT_TYPE, (String docString) -> docString));
     }
 
     public void defineDocStringType(DocStringType docStringType) {
-        DocStringType byContentType = docStringTypesByContentType.get(docStringType.getContentType());
-        if (byContentType != null) {
-            throw createDuplicateTypeException(byContentType, docStringType);
+        if (docStringTypes.containsKey(docStringType.getContentType()) &&
+                docStringTypes.get(docStringType.getContentType()).containsKey(docStringType.getType())) {
+            throw createDuplicateTypeException(
+                docStringTypes.get(docStringType.getContentType()).get(docStringType.getType()),
+                docStringType);
         }
-        DocStringType byClass = docStringTypesByType.get(docStringType.getType());
-        if (byClass != null) {
-            throw createDuplicateTypeException(byClass, docStringType);
-
+        Map<Type, DocStringType> map;
+        if (docStringTypes.containsKey(docStringType.getContentType())) {
+            map = docStringTypes.get(docStringType.getContentType());
+        } else {
+            map = new HashMap<>();
         }
-        docStringTypesByContentType.put(docStringType.getContentType(), docStringType);
-        docStringTypesByType.put(docStringType.getType(), docStringType);
+        map.put(docStringType.getType(), docStringType);
+        docStringTypes.put(docStringType.getContentType(), map);
```

Comment:
Consider using `Map.computeIfAbsent` to simplify section.

https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#computeIfAbsent-K-java.util.function.Function-